src/Event/EuLoginEventSubscriber.php

```diff
@@ -68,16 +68,22 @@ public static function getSubscribedEvents(): array {
    */
   public function generateEmail(CasPreRegisterEvent $event): void {
     $attributes = $event->getCasPropertyBag()->getAttributes();
-    if (!empty($attributes['mail'])) {
-      $event->setPropertyValue('mail', $attributes['mail']);
+    if (!empty($attributes['email'])) {
+      $event->setPropertyValue('mail', $attributes['email']);
     }
-
-    if (!empty($attributes['authenticationFactors'])) {
-      $authFactors = $attributes['authenticationFactors'];
-      if (isset($authFactors['moniker'])) {
-        $event->setPropertyValue('mail', $authFactors['moniker']);
```

Comment:
Are you sure this is correct? Before, the email was coming inside `moniker` attributes. How come this changed? Is it the `userDetails` that makes this change?

Reply:
The problem with this snippet is that its not accurate: moniker is the string used to log in, be it an email or a username. This lead to wrong information being stored in the database. With the userDetails enabled, the email always comes with the "email" property. I could add an exception if no email is found to simply prevent a user from being created, would this make sense?